src/fragmentarium/domain/text.ts

```diff
@@ -140,21 +140,147 @@ export type Token =
   | Enclosure
   | Tabulation
 
-export type Line = RulingDollarLine | LineBase
+export type Line =
+  | TextLine
+  | EmptyLine
+  | LooseDollarLine
+  | ImageDollarLine
+  | RulingDollarLine
+  | SealDollarLine
+  | StateDollarLine
+  | SealAtLine
+  | HeadingAtLine
+  | ColumnAtLine
+  | DiscourseAtLine
+  | SurfaceAtLine
+  | ObjectAtLine
+  | DivisionAtLine
+  | CompositeAtLine
 
 export interface LineBase {
   readonly type: string
   readonly prefix: string
   readonly content: ReadonlyArray<Token>
 }
 
+export interface EmptyLine extends LineBase {
+  readonly type: 'EmptyLine'
+}
+export interface LineNumber {
+  readonly number: number
+  readonly hasPrime: false
+  readonly prefixModifier?: string
+  readonly suffixModifier?: string
+}
+export interface TextLine extends LineBase {
+  readonly type: 'TextLine'
+  readonly lineNumber?: LineNumber
+}
+
+export interface ControlLines extends LineBase {
```

Comment:
Why is this plural? We should try come up with a better name control line is a control line, but at- and dollar-lines are separate now.

Reply:
No changes, not resolved.